### ISSUE TITLE
Make `mode()` a runtime decision

### ISSUE THREAD

**Issue by rrahn** [seqan/seqan3#1436]
(user feedback)

Make it a hybrid configuration, i.e. it can be either set at runtime or at compile time if known already. 
See https://github.com/seqan/seqan3/blob/master/include/seqan3/alignment/configuration/align_config_aligned_ends.hpp as a reference to implement both using almost the same interface.
Make the configuration options enums and then in addition define integral_constant values from the enums. Then the config can be called from the enum and the enum can be changed at runtime, or if the configuration is already known one can use the variable with the encoded enum.

```cpp
enum struct find : uint8_t
{
    best
};
inline constexpr std::integral_constant<find, find::best> find_best{};
...
search_cfg::mode(find::best)  // uses enum
search_cfg::mode(find_best)  // uses static type info
```

- [ ] think about a better name for find
- [ ] adapt the configuration to use enum
- [ ] tests
- [ ] adapt algorithm config
- [ ] adapt docu (API, tutorial)

**Comment by h-2** [comment]
I think I figured out a way we can have nice hybrid options. Have a look at this:

```cpp
enum class select
{
    all,
    all_best,
    best
};

template <select s>
struct static_conf
{
    static constexpr select value = s;
};

struct dynamic_conf
{
    select value;
};

constexpr dynamic_conf mode(select s)
{
    return dynamic_conf{s};
}

template <select s>
constexpr static_conf<s> mode()
{
    return {};
}

int main()
{
    auto dyna   = mode(select::all);
    auto stati  = mode<select::all>;
}
```

`dyna` is a dynamic config, but `stati` is not a static config, it's a function pointer. To make this work the `operator|` that is defined for the config elements would have to get an overload that takes functions pointers of functions whose return type is a specialisation of a static conf. It would then forward to the `operator|` with a default constructed value of that type.

**Comment by h-2** [comment]
BTW: for the problem at hand an enum cannot be used, because `strata` is a valid mode and that needs an argument . I would propose something like this:

```cpp

struct search_mode
{
    static constexpr int32_t all = -1;
    static constexpr int32_t all_best = -2;
    static constexpr int32_t best = -3;

    using strata = int32_t;
};

auto m = search_mode::all;
auto m2 = search_mode::strata{2};
```
`int32_t` can be replaced with a (non-public) strong type.

**Comment by rrahn** [comment]
Well be implemented as part of seqan/seqan3#65
